include/mbgl/style/filter.hpp

```diff
@@ -258,19 +268,15 @@ using FilterBase = variant<
     class IdentifierInFilter,
     class IdentifierNotInFilter,
     class HasIdentifierFilter,
-    class NotHasIdentifierFilter>;
+    class NotHasIdentifierFilter,
+    class ExpressionFilter>;
 
 class Filter : public FilterBase {
 public:
     using FilterBase::FilterBase;
 
-    bool operator()(const Feature&) const;
-
-    template <class GeometryTileFeature>
     bool operator()(const GeometryTileFeature&) const;
-
-    template <class PropertyAccessor>
-    bool operator()(FeatureType type, optional<FeatureIdentifier> id, PropertyAccessor accessor) const;
+    bool operator()(expression::EvaluationContext context) const;
```

Comment:
Should we remove the `bool operator()(const GeometryTileFeature&) const` overload?

Reply:
Absolutely 👍